app/services/site_setting/update.rb

```diff
@@ -47,21 +54,9 @@ def current_user_is_admin(guardian:)
     guardian.is_admin?
   end
 
-  def setting_is_shadowed_globally(params:)
-    !SiteSetting.shadowed_settings.include?(params.setting_name)
-  end
-
-  def setting_is_visible(params:, options:)
-    options.allow_changing_hidden || !SiteSetting.hidden_settings.include?(params.setting_name)
-  end
-
-  def setting_is_configurable(params:)
-    return true if !SiteSetting.plugins[params.setting_name]
-
-    Discourse.plugins_by_name[SiteSetting.plugins[params.setting_name]].configurable?
-  end
-
   def save(params:, guardian:)
-    SiteSetting.set_and_log(params.setting_name, params.new_value, guardian.user)
+    params.settings.each do |setting_name, value|
```

Comment:
I think a small improvement would be to run this in a transaction block, so either all settings save or none do